### PR TITLE
Add option to skip initial calibration and use zero offsets

### PR DESCRIPTION
**Pull request by prashanthr05**
This PR addresses https://github.com/robotology/whole-body-estimators/issues/67.

The idl files were autogenerated due to addition of parameter  `skipResetFTSensorOffset` to ` idl/wholeBodyDynamicsSettings/wholeBodyDynamicsSettings.thrift`. The autogenerated files are also committed along with this PR.

Since by default the parameter `skipResetOffset`  is set to `false` (as per the regular workflow), only configuration related to simulation was modified by this PR.   `devices/wholeBodyDynamics/app/wholebodydynamics-icub-external-six-fts-sim.xml`

### CODE REVIEW

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 1]
```diff
@@ -942,7 +961,7 @@ bool WholeBodyDynamicsDevice::applyLPFSettingsFromConfig(const yarp::os::Propert
     {        
         return false;
     }
-    
+
```
Comment (traversaro) [inline devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:964]:
```suggestion

```

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 2]
```diff
@@ -1363,8 +1382,17 @@ bool WholeBodyDynamicsDevice::attachAll(const PolyDriverList& p)
...
     ok = ok && this->attachAllFTs(p);
     ok = ok && this->attachAllIMUs(p);
-
-    ok = ok && this->setupCalibrationWithExternalWrenchOnOneFrame("base_link",100);
+    
+    if (settings.skipResetFTSensorOffset)
+    {
+        this->setFTSensorOffsetsToZero();
+        validOffsetAvailable = true;
+        ok = ok && true;
```
Comment (traversaro) [inline devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:1390]: It is not really clear what is the  point of this line.

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 3]
```diff
@@ -924,6 +924,25 @@ bool WholeBodyDynamicsDevice::loadSettingsFromConfig(os::Searchable& config)
     if (settings.useJointVelocity) { yInfo() << "wholeBodyDynamics: jointVelFilterCutoffInHz: " <<   settings.jointVelFilterCutoffInHz << " Hz."; }
     if (settings.useJointAcceleration) { yInfo() << "wholeBodyDynamics: jointAccFilterCutoffInHz: " <<   settings.jointAccFilterCutoffInHz << " Hz."; }
     
+    if ( !(prop.check("skipResetOffset") && prop.find("skipResetOffset").isBool()) )
+    {
+        yWarning() << "wholeBodyDynamics: skipResetOffset bool parameter missing.";
+        yWarning() << "wholeBodyDynamics: setting skipResetOffset to the default value of false, please remember to resetOffset through the rpc port.";
```
Comment (traversaro) [inline devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:930]: This is misleading. On the real robot, you don't have to remember to call `resetOffset` .

Comment (prashanthr05) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:930]: Noted. I will remove that line.

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp (hunk 2 above)
Comment (prashanthr05) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:1390]: I added that redundant code to debug and forgot to remove.

**Comment by traversaro** [comment]
Thanks! Minor comments: to be honest, the parameter name  `skipResetFTSensorOffset` probably make sense to existing users that called `resetOffets` for thousands of times in their life, but I think it is misleading for new users, as it does suggest that the "FT Sensor offset reset" is skipped, while this option does exactly the opposite: is uses zero FT sensor offsets (so, "reset" offset) at the device startup, instead of performing calibration. For this reason, I think that calling something like `startWithZeroFTSensorOffsets` or similar (suggestions are welcome) would be more clear. Furthermore, also the description is not really clear: "This allows to skip manual calling of resetOffsets for removeing offsets on FT sensors". I think it  would me more clear to first have a place were we state (also in the general description before this parameter) that at the beginning the device estimates the offset of the FT sensors, unless the option `startWithZeroFTSensorOffsets` is passed. 

Furthermore note that a similar functionality (let user specify user-defined offsets, not just 0) is also provided in https://github.com/robotology/whole-body-estimators/pull/45, so we should consider if  the two options can coexist (probably yes, `startWithZeroFTSensorOffsets` can be an easy option to set all the FT sensor offset to zero (useful in simulation), while the option https://github.com/robotology/whole-body-estimators/pull/45 can be used for more complex settings in which the offset need to be manually inserted. 


@GiulioRomualdi @Giulero @gabrielenava @HosameldinMohamed

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp (hunk 3 above)
Comment (prashanthr05) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: Addressed in https://github.com/robotology/whole-body-estimators/pull/72/commits/b495053958d918d01711010f7e32f1ac651fc29c

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp (hunk 2 above)
Comment (prashanthr05) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: Addressed in https://github.com/robotology/whole-body-estimators/pull/72/commits/b495053958d918d01711010f7e32f1ac651fc29c

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp (hunk 1 above)
Comment (prashanthr05) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: Addressed in https://github.com/robotology/whole-body-estimators/pull/72/commits/b495053958d918d01711010f7e32f1ac651fc29c

**Comment by prashanthr05** [comment]
>  I think that calling something like startWithZeroFTSensorOffsets or similar (suggestions are welcome) would be more clear.

Agreed. I think `startWithZeroFTSensorOffsets` is clear. But I will wait a bit to understand if we get some inputs from other users.

> Furthermore note that a similar functionality (let user specify user-defined offsets, not just 0) is also provided in #45, so we should consider if the two options can coexist (probably yes, startWithZeroFTSensorOffsets can be an easy option to set all the FT sensor offset to zero (useful in simulation), while the option #45 can be used for more complex settings in which the offset need to be manually inserted.

I revisited that PR and noticed that. Indeed, it allows the user to add an `FT_OFFSET` group to specify the custom offsets. I hope we will not be crowding the source code too much by adding a redundant feature (`startWithZeroFTSensorOffsets`).


However, if its agreed to have both the options so that they can be used without any confusion. I will rename the parameter to what is agreed upon. Otherwise, we can close this PR.

**Comment by traversaro** [comment]
> However, if its agreed to have both the options so that they can be used without any confusion, I will rename the parameter to what is agreed upon. 

I agree that having both options make sense, especially for simulation users so that they do not need to add a complex `FT_OFFSET` group when they can just add a simple  option.

**Comment by traversaro** [comment]
cc @lrapetti @diegoferigo

**Comment by prashanthr05** [comment]
> Agreed. I think startWithZeroFTSensorOffsets is clear. But I will wait a bit to understand if we get some inputs from other users.

I will proceed to rename the option with `startWithZeroFTSensorOffsets`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+### Added
+- Added 'skipResetOffset' to disable/enable bypassing of resetOffset rpc call for setting FT sensor offsets to zero. (See [!72](https://github.com/robotology/whole-body-estimators/pull/72)).
```
Comment (traversaro) [inline CHANGELOG.md:9]:
```suggestion
- Added 'startWithZeroFTSensorOffsets' to disable/enable bypassing of resetOffset rpc call for setting FT sensor offsets to zero. (See [!72](https://github.com/robotology/whole-body-estimators/pull/72)).
```

**Comment by traversaro** [comment]
Approved, but check the CHANGELOG suggestion.

File: CHANGELOG.md (hunk 4 above)
Comment (prashanthr05) [reply CHANGELOG.md]: Addressed in https://github.com/robotology/whole-body-estimators/pull/72/commits/52813c1dfd0470723a4703d4ff165b81e76bcea6

**Comment by traversaro** [comment]
Perfect!

**Pull request merged**